lib/rules/require-dot-notation.js

```diff
@@ -1,11 +1,16 @@
 /**
  * Requires member expressions to use dot notation when possible
  *
- * Types: `Boolean` or `String`
+ * Types: `Boolean` or `Object`
  *
  * Values:
  *  - `true`
- *  - `"except_snake_case"` allow quoted snake cased identifiers
```

Comment:
Even though a lot of rule values are deprecated, we do not removing them, leaving this task for 3.0